homeassistant/components/deconz/sensor.py

```diff
@@ -67,171 +70,163 @@
 ATTR_EVENT_ID = "event_id"
 
 
+T = TypeVar(
+    "T",
+    AirQuality,
+    Consumption,
+    Daylight,
+    GenericStatus,
+    Humidity,
+    LightLevel,
+    Power,
+    Pressure,
+    Temperature,
+    Time,
+    PydeconzSensorBase,
+)
+
+
 @dataclass
-class DeconzSensorDescriptionMixin:
+class DeconzSensorDescriptionMixin(Generic[T]):
     """Required values when describing secondary sensor attributes."""
 
+    isinstance_fn: Callable[[T], bool]
     update_key: str
-    value_fn: Callable[[SensorResources], float | int | str | None]
+    value_fn: Callable[[T], datetime | StateType]
 
 
 @dataclass
 class DeconzSensorDescription(
-    SensorEntityDescription,
-    DeconzSensorDescriptionMixin,
+    SensorEntityDescription, DeconzSensorDescriptionMixin[T], Generic[T]
```

Comment:
`DeconzSensorDescriptionMixin[T]` is already a generic, so you can strip `Generic[T]`.

Reply:
Thanks!

Reply:
Resolved with https://github.com/home-assistant/core/pull/79272